src/distribution.cpp

```diff
@@ -29,27 +29,67 @@ Discrete::Discrete(pugi::xml_node node)
   std::copy(params.begin(), params.begin() + n / 2, std::back_inserter(x_));
   std::copy(params.begin() + n / 2, params.end(), std::back_inserter(p_));
 
-  normalize();
+  // Initialize alias tables
+  init_alias();
 }
 
 Discrete::Discrete(const double* x, const double* p, int n)
   : x_ {x, x + n}, p_ {p, p + n}
 {
+  // Initialize alias tables
+  init_alias();
+}
+
+void Discrete::init_alias() {
   normalize();
+
+  // Vectors for large and small probabilities based on 1/n
```

Comment:
I would expand the comment here to explain that the method used to fill `prob_` and `alias_` is based on the method by Vose and provide the DOI: https://doi.org/10.1109/32.92917